### PR TITLE
Add tests for `skipPoll`, and fix `$not` and `$nor` while at it

### PR DESCRIPTION
**Pull request by avital**
This pull request adds a large suite of tests for many of the queries
that `skipPoll` knows how to deal with.

While running these tests, I found a bug dealing with queries with
`$not` and `$nor`. This pull requests includes a fix for those queries.

### CODE REVIEW

**Comment by avital** [comment]
Updated the PR to no longer use underscore, and this helped me simplify the tests and make them faster as well.

File: index.js [hunk 1]
```diff
@@ -774,11 +774,17 @@ ShareDbMongo.prototype.canPollDoc = function(collectionName, query) {
 // Return true to avoid polling if there is no possibility that an op could
 // affect a query's results
 ShareDbMongo.prototype.skipPoll = function(collectionName, id, op, query) {
-  // Livedb is in charge of doing the validation of ops, so at this point we
+  // Sharedb is in charge of doing the validation of ops, so at this point we
```
Comment (nateps) [inline index.js]: ShareDB is the correct capitalization

File: test/test_skip_poll.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            {op: [{p: ['x', 'y', 'z', 1, 2, 3, 'd'], dummyOp: 1}]},
+            query, !has(fields, 'x'));
+
+          // modifying empty path (could theoretically modify any part
+          // of the document)
+          assertSkips(
+            {op: [{p: [], dummyOp: 1}]},
+            query, false);
+
+          // multiple operations at once
```
Comment (avital) [inline test/test_skip_poll.js]: make into separate its

File: test/queries_to_fields.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+//
+// Exports a an array of objects with `query` and `fields` properties, e.g.:
+//
+// [
+//   {query: {a: {$ne: 1}, b: 2}, fields: ['a', 'b']},
+//   (...)
+// ]
+
+// set of base queries with top-level fields
+var BASE = [
```
Comment (avital) [inline test/queries_to_fields.js]: put this in setup block for describe()

**Pull request closed without merging**